aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -211,7 +211,7 @@ Tensor empty_like(
     }
   }
 
-  if (self.is_quantized()) {
+  if (self.is_quantized() && options.dtype() == self.dtype()) {
     // We could check if dtype is still quantized?  But then should we shift/scale
     // the q_zero_point / q_scale or not?
     TORCH_CHECK(!options.has_dtype() || options.dtype() == self.dtype(),
```

Comment:
Why is this restriction in place?

Reply:
I think for the case where the self.dtype() is quantized and options.dtype() is non-quantized, for example, when we apply:
```
auto buffer = at::zeros_like(output, output.options().dtype(at::kInt));
```
where `output` is a quantized tensor with `kQUInt8` dtype, we should still be able to create an empty tensor with
```
at::empty(self.sizes(), options, use_memory_format);
```
(Note that we shouldn't use `at::_empty_affine_quantized` or `at::_empty_per_channel_affine_quantized` as options.dtype() is non-quantized).

With the previous version of code before this PR, we just report the errors.